src/DevBetterWeb.Core/Services/MemberAddBillingActivityService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Threading.Tasks;
+using DevBetterWeb.Core.Entities;
+using DevBetterWeb.Core.Interfaces;
+using DevBetterWeb.Core.Specs;
+
+namespace DevBetterWeb.Core.Services
+{
+  public class MemberAddBillingActivityService : IMemberAddBillingActivityService
+  {
+    private readonly IUserLookupService _userLookup;
+    private readonly IMemberSubscriptionPeriodCalculationsService _memberSubscriptionPeriodCalculationsService;
+    private readonly IRepository _repository;
+
+    public MemberAddBillingActivityService(IUserLookupService userLookup,
+      IMemberSubscriptionPeriodCalculationsService memberSubscriptionPeriodCalculationsService,
+      IRepository repository)
+    {
+      _userLookup = userLookup;
+      _memberSubscriptionPeriodCalculationsService = memberSubscriptionPeriodCalculationsService;
+      _repository = repository;
+    }
+
+    public async Task AddSubscriptionCreationBillingActivity(string email, decimal amount)
+    {
+      var member = await GetMemberByEmailAsync(email);
+
+      var message = $"{member.UserFullName()} subscribed to DevBetter";
```

Comment:
As we discussed, put more info into the value object so that you can build the message based on its properties.

In member activity, you don't want the member full name, but in admin activity reports you will. So you need to know in the value object: memberfullname, activityverbpasttense, subpscriptionplanName and possibly the billing period. A message might look like this:

```
$"{MemberFullname} {ActivityVerbPastTense} to {PlanName} billed {BillingPeriod}."
```